lightningd/channel_control.c

```diff
@@ -56,6 +56,13 @@ void channel_update_feerates(struct lightningd *ld, const struct channel *channe
 	else
 		min_feerate = feerate_min(ld, NULL);
 	max_feerate = feerate_max(ld, NULL);
+	/* The channel opener should use a slightly higher than minimal feerate
+	 * in order to avoid excessive feerate disagreements */
+	if (channel->opener == LOCAL) {
+		min_feerate += ld->config.feerate_offset;
+		max_feerate += ld->config.feerate_offset;
```

Comment:
What effect do max and min have if we're the opener?  I expected you to simple add the offset to feerate, capped at max_feerate.

Reply:
Hmm, I am not sure if this solution offers any benefits because we can pay extra fee? 

The fee estimation from the backend is quite chaotic. There are backends like Neutrino that use external APIs, and there are other backends that perform different calculations to estimate the fee because they have copied the core Lightning code incorrectly. So the diff between our and the remote can be also `our_fee: 250 vs peer_fee: 240`. 

I was considering addressing this issue since I have a log of nodes experiencing this problem. So, why not try the following approach:

Estimate a percentage that is 5% or 10% of margin (the default value can be adjust from future bug report, lets consider to start with `5%`):

- if our fee is lower than the fee that other peer propose, check:  
     - if the `remote_fee > local_fee + percentage` disable the channel;
     - if the `remote_fee <= local_fee + percentage` override our fee with the `remote_fee`

In the case below, the user sets the percentage through a ‘tolerance’ field, which allows for solving future problems as they arise.

With your formula `max_feerate += ld->config.feerate_offset;` we are going to add the `feerate_offset` to our feerate but this can be wrong in the case that other node took this `feerate_offset` as a too hight fee. LDK for instance force close the channel iirc if there is a fee disagreement

Does this make sense?

Reply:
This does add an extra marginal fee, but the alternative can be frequent forced disconnects from certain peers.

>if the remote_fee <= local_fee + percentage override our fee with the remote_fee

So this PR doesn't address the problem where your peer proposes too low of a fee.  I think if that happens, it's dangerous to use a lower feerate (but you're already welcome to override that on a per-peer basis.)  Maybe more nuanced behavior there is worth exploring.  But I don't want to touch this one for now.

> if the remote_fee > local_fee + percentage disable the channel;

We already have the min and max range we'll accept and the maximum is already very generous, but I think we should stick to using it as the maximum limit.

So what does this actually address? Really just the cases where the local node is the channel opener (and sets the feerate) and also may lowball the fee relative to the remote node.  I'll quote the spec:
>The node responsible for paying the Bitcoin fee:
>
>    SHOULD send update_fee to ensure the current fee rate is sufficient (by a significant margin) for timely processing of the commitment transaction.

So our problem is we (as the channel opener) currently set the update_fee to the minimum we consider reasonable (but not by a significant margin.)  I don't know the best way to calculate that margin. From the data I've collected a margin of 5sats/kW would take care of the minor fee disagreements (rounding or timing errors) where the local and remote node are off by one. For some node pairs this is the most common problem.  For other node pairs, like you suggest, they may be farther apart and need quite a lot more, I just haven't determined a good value there.  I'm open to using a percentage if that seems more reasonable based on the data.  (I could also see an argument for a fixed offset default in the range of 70-100. if we think 5 wouldn't catch enough disagreements)

If you want to run [this script](https://gist.github.com/endothermicdev/1542d3fdac7d5311b08394f5fcaeb402) against your logs and try swapping line 118 for 119, maybe we could find an appropriate percentage and then take that approach?

Reply:
> What effect do max and min have if we're the opener? I expected you to simple add the offset to feerate, capped at max_feerate.

Done

Reply:
>We already have the min and max range we'll accept and the maximum is already very generous, but I think we should stick to using it as the maximum limit.

Yes, I know and it is a monkey one because forces the user to look at the log and disable the fee checking, if you are a user with an external application you are not able to do it.

My point is that now that we are having the problem in the opposite way, we could move the checking fee logic into a single function and unify behaviour that makes it a little bit user-friendly

IMHO instead of putting this change in we should do some more research on how to make this ignore fee logic better and maybe unify it across implementations. However, it is not a problem till no one complains